kong/runloop/plugins_iterator.lua

```diff
@@ -539,102 +427,31 @@ function PluginsIterator.new(version)
 
       plugins[name] = true
 
-      local combo_key = (plugin.route    and 1 or 0)
-                      + (plugin.service  and 2 or 0)
-                      + (plugin.consumer and 4 or 0)
-
-      local cfg
-      if combo_key == COMBO_GLOBAL then
-        cfg = get_plugin_config(plugin, name, ws_id)
-        if cfg then
-          globals[name] = cfg
-        end
+      -- Retrieve route_id, service_id, and consumer_id from the plugin object, if they exist
+      local route_id = plugin.route and plugin.route.id
+      local service_id = plugin.service and plugin.service.id
+      local consumer_id = plugin.consumer and plugin.consumer.id
+
+      -- Get the plugin configuration for the specified workspace (ws_id)
+      local cfg = get_plugin_config(plugin, name, ws_id)
+      -- Determine if the plugin configuration is global (i.e., not tied to any route, service, or consumer)
+      local is_global = not route_id and not service_id and not consumer_id
+      if is_global then
+        -- Store the global configuration for the plugin in the 'globals' table
+        globals[name] = cfg
```

Comment:
Is `globals[name]` redundant? I.e. would it always have the same content as `combos[name]["::"]`? `globals` is exposed outside of this module though so maybe we are intentionally keeping the tables separate.

Reply:
I kept the behavior close to the original implementation before we _fully_ understand the behavior of `global` plugins. See https://github.com/Kong/kong-ee/pull/5168